modules/sdk-coin-sol/test/unit/sol.ts

```diff
@@ -1482,5 +1497,27 @@ describe('SOL:', function () {
       should.equal(durableNonceTxnJson.feePayer, testData.SolInputData.pubKey);
       should.equal(durableNonceTxnJson.numSignatures, testData.SolInputData.durableNonceSignatures);
     });
+
+    it('should recover a txn for unsigned sweep recoveries', async function () {
+      // Unsigned Sweep Recovery
+      const unsignedSweepTxn = await basecoin.recover({
+        bitgoKey: testData.keys.bitgoKey,
+        recoveryDestination: testData.keys.destinationPubKey,
+        walletPassphrase: testData.keys.walletPassword,
+        durableNoncePK: testData.keys.durableNoncePubKey,
+        durableNonceSK: testData.keys.durableNoncePrivKey,
```

Comment:
Since we're dealing with a few optional properties, it would be good to add some tests with edge cases (unexpected input, or undefined inputs) to ensure the overall method fails well.